### PR TITLE
Hardware intrinsics tests in new style

### PR DESCRIPTION
**Pull request by davidwrighton**
Port all of the vector tests to the new xunit style specification
- Test disables need to use the attribute based form
- All tests are run on all architectures (This wasn't happening in the old system, but hopefully with test striping the need for this has gone away.)
- The xunit wrapper generator generated methods which are too big for the merged assembly as designed earlier. I've broken that up into chunks of 100 methods each which seems to have solved that problem.
- The xunit wrapper generator and runner have issues with running this many tests under GCStress, I've implemented a striping scheme to allow these tests to run across more machines when running under GCStress.

### CODE REVIEW

**Comment by davidwrighton** [comment]
@lambdageek I'm hitting an error on the llvmfullaot and llvmaot runs where the message is `Ran out of trampolines of type 0 in '/datadisks/disk1/work/BF790A62/p/System.Private.CoreLib.dll' (limit 10000)`. From what I can see, this is something where there is some sort of parameter available to increase the limit at build time. Could you give me some advice on how to do so?

**Comment by davidwrighton** [comment]
@lambdageek So, I doubled that number and now I see very odd failures. Could you take a look at what happened? For llvmaot mode, I see that the tests appear to pass, and generate a testresults.xml file, but something in our test infra thinks that test file is invalid, and for the llvmfullaot mode it seems to crash unexpectedly while executing.

**Comment by lambdageek** [comment]
llvmfullaot on x64 + arm64 is failing with
```
Ran out of trampolines of type 3 in '/root/helix/work/correlation/System.Private.CoreLib.dll' (limit 4400)
```

type 3 is ... TBD.  but this should be fixable.

---


llvmaot is failing with some weird error

@steveisok The tests here are failing on mono's LLVM lanes in a very weird way - in the logs you can see the tests failing, but then near the end the test says "PASSED" but then we have trouble generating a test results report or something:

```
2:40:35.572 Failed test: global::JIT.HardwareIntrinsics.X86._Sse2.Program.UnpackHighInt64()
...
ctual: 100
END EXECUTION - PASSED
+ export _commandExitCode=0
+ /usr/bin/python3 /datadisks/disk1/work/9FAD08B0/p/reporter/run.py https://dev.azure.com/dnceng-public/ public 789902 eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsIng1dCI6Im9PdmN6NU1fN3AtSGpJS2xGWHo5M3VfVjBabyJ9.eyJuYW1laWQiOiJjNzczZjJjMi01MTIwLTQyMDctYWZlMi1hZmFmMzVhOGJjMGEiLCJzY3AiOiJhcHBfdG9rZW4iLCJhdWkiOiIwMjQ4YjJjOS02YzFjLTQ3ZTItODU2YS0wNTc4ZjZhM2I1NmIiLCJzaWQiOiJhNTc3NDBiMS1mMGVkLTQ1YWItYmUxMy1mYzdiYjQ4MDQxOWYiLCJCdWlsZElkIjoiY2JiMTgyNjEtYzQ4Zi00YWJiLTg2NTEtOGNkY2I1NDc0NjQ5OzM5Mzc0IiwicHBpZCI6InZzdGZzOi8vL0J1aWxkL0J1aWxkLzM5Mzc0Iiwib3JjaGlkIjoiZTUyZmM1MTUtMDE5Zi00MDU0LTkzN2YtOTI3NmE4MTY4MjFlLnJ1bl90ZXN0X3AwX21vbm9sbHZtYW90X19saW51eF94NjRfcmVsZWFzZS5fX2RlZmF1bHQiLCJyZXBvSWRzIjoiIiwiaXNzIjoiYXBwLnZzdG9rZW4udmlzdWFsc3R1ZGlvLmNvbSIsImF1ZCI6ImFwcC52c3Rva2VuLnZpc3VhbHN0dWRpby5jb218dnNvOjZmY2M5MmU1LTczYTctNGY4OC04ZDEzLWQ5MDQ1YjQ1ZmIyNyIsIm5iZiI6MTY2NDg0NzAyNCwiZXhwIjoxNjY0ODYwMjI0fQ.ZG2FgLtSbCG3LfUPZQTOBDLbT-x4yAYpELk4EjKyzhtC6VLGayvj5_yux4a79eTFqiEXqpkscuNYUMMtw1-Gzq4Auh5lnB9SU8TvJ0-pRrJ3w3rmUbWHz1ryNgSHs3kXTr-TF9iRl7Cxj9HN7wfRyiG7kXsoiR2zULwRH0atgrpJkNpz45hEPMW-o4kBx9yZY90uVKtxPNMoknRijjmn53y0tP3HVJpacJpTlCSJAN-D1u-oOeJab7VbzvDQ4Re2ZoKCTo_3N0s95U-CwXaYV-BvSwVy52vbLMfoFXwVMjGs4so0cQ6xZr9gUlcGH6d9qLrcKCDZDBPfKeImR370RA
2022-10-04T02:40:38.344Z	INFO   	run.py	run(48)	main	Beginning reading of test results.
2022-10-04T02:40:38.344Z	INFO   	run.py	__init__(42)	read_results	Searching '/datadisks/disk1/work/9FAD08B0/w/A8D30910/e' for test results files
2022-10-04T02:40:38.345Z	INFO   	run.py	__init__(48)	read_results	Found results file /datadisks/disk1/work/9FAD08B0/w/A8D30910/e/JIT/HardwareIntrinsics/HardwareIntrinsics_ro/HardwareIntrinsics_ro.testResults.xml with format xunit
Traceback (most recent call last):
  File "/datadisks/disk1/work/9FAD08B0/p/reporter/run.py", line 68, in <module>
    main()
  File "/datadisks/disk1/work/9FAD08B0/p/reporter/run.py", line 65, in main
    reporter.report_results(all_results)
  File "/etc/helix/scripts/helix/test_reporting/packing_test_reporter.py", line 27, in report_results
    results = [x for x in results if x]
  File "/etc/helix/scripts/helix/test_reporting/packing_test_reporter.py", line 27, in <listcomp>
    results = [x for x in results if x]
  File "/datadisks/disk1/work/9FAD08B0/p/reporter/test_results_reader/__init__.py", line 51, in read_results
    for result in file_results:
  File "/datadisks/disk1/work/9FAD08B0/p/reporter/formats/xunit.py", line 65, in read_results
    failure_message = _unescape_xunit_message(message_element.text)
  File "/datadisks/disk1/work/9FAD08B0/p/reporter/formats/xunit.py", line 28, in _unescape_xunit_message
    return re.sub(r'\\x[0-9a-fA-F][0-9a-fA-F][0-9a-fA-F]?[0-9a-fA-F]?|\\[^x]', bs, value)
  File "/usr/lib/python3.6/re.py", line 191, in sub
    return _compile(pattern, flags).sub(repl, string, count)
TypeError: expected string or bytes-like object
+ exit 1
```

**Comment by davidwrighton** [comment]
@lambdageek  I've had a bit of time to look into what's going on here, and as I don't understand the runtime environment of mono, I'm not making appreciable progress. Could you arrange for someone from the mono team to take a look. I'm available over teams for any questions about this, but I don't think I'll be able to actually make this work without direct assistance.

**Comment by steveisok** [comment]
@davidwrighton Can you resolve the conflict and kick another build off?

**Comment by davidwrighton** [comment]
@steveisok  I've resolved it. Sorry about the delay.

**Comment by davidwrighton** [comment]
@steveisok I see that this is now also failing in some coreclr issue, but the mono issue is still here.

**Comment by steveisok** [comment]
@lambdageek I think whatever infra hang up was happening before is no longer.  There are a few crashes and test failures that may or may not be related to the PR.

**Comment by lambdageek** [comment]
@steveisok now lanes like `Build mono llvmfullaot Pri0 Runtime Tests Run Linux arm64 release failed` look like the helix jobs are timing out, don't they? i see it running for 2.5 hours and then it dies, but the backtrace looks like something is killing the job.

**Comment by lambdageek** [comment]
`Vector64.op_Inequality<Byte>(Vector64<Byte>, Vector64<Byte>): RunClassLclFldScenario` on `mono minijit Pri0 Runtime Tests Run OSX x64 release` seems like a real failure.

same with `mono monointerpreter Pri0 Runtime Tests Run OSX x64 release`

```
Running test: global::JIT.HardwareIntrinsics.General._Vector64_1.Program.op_InequalitySByte()
Beginning scenario: RunBasicScenario_UnsafeRead
Vector64.op_Inequality<SByte>(Vector64<SByte>, Vector64<SByte>): RunBasicScenario_UnsafeRead failed:
    left: (8, 121, 22, 85, 58, 79, 84, 42)
   right: (114, 5, 115, 85, 37, 94, 9, 108)
  result: (False)

Beginning scenario: RunReflectionScenario_UnsafeRead
Vector64.op_Inequality<SByte>(Vector64<SByte>, Vector64<SByte>): RunReflectionScenario_UnsafeRead failed:
    left: (8, 121, 22, 85, 58, 79, 84, 42)
   right: (114, 5, 115, 85, 37, 94, 9, 108)
  result: (False)

Beginning scenario: RunClsVarScenario
Beginning scenario: RunLclVarScenario_UnsafeRead
Vector64.op_Inequality<SByte>(Vector64<SByte>, Vector64<SByte>): RunLclVarScenario_UnsafeRead failed:
    left: (8, 121, 22, 85, 58, 79, 84, 42)
   right: (114, 5, 115, 85, 37, 94, 9, 108)
  result: (False)

Beginning scenario: RunClassLclFldScenario
Beginning scenario: RunClassFldScenario
Beginning scenario: RunStructLclFldScenario
Vector64.op_Inequality<SByte>(Vector64<SByte>, Vector64<SByte>): RunStructLclFldScenario failed:
    left: (106, 54, 30, 41, 45, 19, 12, 59)
   right: (90, 123, 91, 9, 42, 19, 1, 115)
  result: (False)
``` 

---

on `mono llvmfullaot Pri0 Runtime Tests Run Linux x64 release` some of the failures seem to be because the tests aren't properly excluded:

the `Vector256_1` tests should be excluded for llvmfullaot but aren't https://github.com/dotnet/runtime/blob/9e1350eb6d8d6018d599b7fefbae6805dbd420ea/src/tests/issues.targets#L3067-L3069
also the `X86/Sse42.X64` tests should be excluded for mono but aren't
https://github.com/dotnet/runtime/blob/main/src/tests/issues.targets#L1420-L1422

likewise, the `X86/X86Base/Pause_ro` tests should be excluded on llvmfullaot and llvmaot, but aren't

https://github.com/dotnet/runtime/blob/9e1350eb6d8d6018d599b7fefbae6805dbd420ea/src/tests/issues.targets#L3112-L3114


@davidwrighton @steveisok does the new infrastructure support excluding tests based on RuntimeFlavor and RuntimeVariant?

**Comment by davidwrighton** [comment]
@lambdageek, the excludes are supposed to be done via attributes now. I thought I had caught most of those, but I'll take another look on Monday.

File: src/coreclr/vm/zapsig.cpp [hunk 1]
```diff
@@ -12,6 +12,7 @@
 // ===========================================================================
 
 
+
```
Comment (davidwrighton) [inline src/coreclr/vm/zapsig.cpp:15]: Remove this extra line before checkin. (It's here to convince CI to run the right set of tests.)

File: src/tests/Common/XUnitWrapperLibrary/TestFilter.cs [hunk 2]
```diff
@@ -161,7 +161,7 @@ public TestFilter(string[] filterArgs, HashSet<string>? testExclusionList)
         {
             if (filterString.IndexOfAny(new[] { '!', '(', ')', '~', '=' }) != -1)
             {
-                throw new ArgumentException("Complex test filter expressions are not supported today. The only filters supported today are the simple form supported in 'dotnet test --filter' (substrings of the test's fully qualified name). If further filtering options are desired, file an issue on dotnet/runtime for support.", nameof(filterString));
+                throw new ArgumentException("Complex test filter expressions ar e not supported today. The only filters supported today are the simple form supported in 'dotnet test --filter' (substrings of the test's fully qualified name). If further filtering options are desired, file an issue on dotnet/runtime for support.", nameof(filterString));
```
Comment (davidwrighton) [inline src/tests/Common/XUnitWrapperLibrary/TestFilter.cs:164]: Remove unnecessary space.

File: src/tests/JIT/HardwareIntrinsics/Arm/AdvSimd.Arm64/Abs.Vector128.Double.cs [hunk 3]
```diff
@@ -13,12 +13,14 @@
 using System.Runtime.InteropServices;
 using System.Runtime.Intrinsics;
 using System.Runtime.Intrinsics.Arm;
+using Xunit;
 
-namespace JIT.HardwareIntrinsics.Arm
+namespace JIT.HardwareIntrinsics.Arm._AdvSimd.Arm64
```
Comment (tannergooding) [inline src/tests/JIT/HardwareIntrinsics/Arm/AdvSimd.Arm64/Abs.Vector128.Double.cs:18]: Just wondering, why the need for the underscore here?

**Review by tannergooding — APPROVED** [review]
Changes LGTM. Fairly mechanical in nature

File: src/tests/JIT/HardwareIntrinsics/Arm/AdvSimd.Arm64/Abs.Vector128.Double.cs (hunk 3 above)
Comment (davidwrighton) [reply src/tests/JIT/HardwareIntrinsics/Arm/AdvSimd.Arm64/Abs.Vector128.Double.cs:18]: Its been quite a while, but I think there was a conflict between a namespace name and class name, and I stuffed in an underscore on the namespace to shut it all up.

**Pull request merged**